File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people.sort()
    
    result = 0
    start = 0
    end = len(people)
    
    while end-start >= 1:
        
        #최소 2개 이상 남아있을 때 루프 검사
        if people[end-1] + people[start] <= limit and end-start >= 2:
            start+=1
            end-=1
        else:
            end-=1
        result+=1
        

    return result

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int rideTogether = 0;
        
        Arrays.sort(people);
        
        int left = 0;
        int right = people.length - 1;
        
        while(left < right){
            if(people[left] + people[right] <= limit){ // 둘이 타기에 충분한 경우
                left++;
                right--;
                rideTogether++;
            }else{ // 한명밖에 못타는 경우 -> 무게가 더 많이 나가는 사람을 대체
                right--; 
            }
        }
        
        return people.length - rideTogether;
    }
}